### PR TITLE
feat(v4-docs): update auth provider docs

### PR DESCRIPTION
**Pull request by alicanerdurmaz**
all examples, tutorials, and, references updated to new auth provider

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: documentation/docs/tutorial/5-understanding-authprovider/1-create-authprovider.md [hunk 1]
```diff
@@ -598,43 +684,62 @@ const authProvider: AuthProvider = {
...
 const mockUsers = [{ email: "john@mail.com" }, { email: "jane@mail.com" }];
 
-const authProvider: AuthProvider = {
-    ...
+const authProvider: AuthBindings = {
+    // ---
     register: ({ email }) => {
         const user = mockUsers.find((user) => user.email === email);
 
         if (user) {
```
Comment (alicanerdurmaz) [inline documentation/docs/tutorial/5-understanding-authprovider/1-create-authprovider.md:725]:
```suggestion
        if (!user) {
```

File: documentation/docs/api-reference/core/components/auth/authenticated.md [hunk 2]
```diff
@@ -3,9 +3,9 @@ id: authenticated
 title: <Authenticated>
 ---
 
-`<Authenticated>` is the component form of [`useAuthenticated`][useauthenticated].
+`<Authenticated>` is the component form of [`useIsAuthenticated`][use-is-authenticated].
 
-It internally uses [`useAuthenticated`][useauthenticated]'s return values (`isSuccess`, `isLoading`, and, `isError`) to provide its functionality.
+It internally uses [`useIsAuthenticated`][use-is-authenticated]'s return values (`isSuccess`, `isLoading`, and, `isError`) to provide its functionality.
```
Comment (aliemir) [inline documentation/docs/api-reference/core/components/auth/authenticated.md:8]: Not applicable to the current interface of the `authProvider`. `data` is used mainly for the functionality.

File: documentation/docs/advanced-tutorials/auth/auth0.md [hunk 3]
```diff
@@ -133,30 +133,59 @@ const App = () => {
         return <span>loading...</span>;
     }
 
-    const authProvider: AuthProvider = {
-        login: () => {
-            return Promise.resolve();
+    const authProvider: AuthBindings = {
```
Comment (BatuhanW) [inline documentation/docs/advanced-tutorials/auth/auth0.md:136]: Should we also update import above, from AuthProvider to AuthBindings?

File: documentation/docs/advanced-tutorials/custom-pages.md [hunk 4]
```diff
@@ -166,23 +166,36 @@ import routerProvider from "@pankod/refine-react-router-v6";
 
 import { CustomPage } from "pages/custom-page";
 
-const authProvider: AuthProvider = {
-    login: (params: any) => {
-        if (params.username === "admin") {
-            localStorage.setItem("username", params.username);
-            return Promise.resolve();
+const authProvider: AuthBindings = {
```
Comment (BatuhanW) [inline documentation/docs/advanced-tutorials/custom-pages.md:169]: Same here, I think we should update import above. Right now AFAIK, AuthBindings isn't imported.

File: documentation/docs/advanced-tutorials/data-provider/appwrite.md [hunk 5]
```diff
@@ -60,37 +60,75 @@ import { AuthProvider } from "@pankod/refine-core";
 
 import { account } from "./appwriteClient";
 
-export const authProvider: AuthProvider = {
+const authProvider: AuthBindings = {
```
Comment (BatuhanW) [inline documentation/docs/advanced-tutorials/data-provider/appwrite.md:63]: Same here, let's check ad make sure we import AuthBindings.

File: documentation/docs/advanced-tutorials/data-provider/supabase.md [hunk 6]
```diff
@@ -162,116 +162,218 @@ import { supabaseClient } from "utility";
 const authProvider: AuthProvider = {
```
Comment (BatuhanW) [inline documentation/docs/advanced-tutorials/data-provider/supabase.md:162]: Should we change this to AuthBindings?

File: documentation/docs/advanced-tutorials/multi-tenancy/strapi.md [hunk 7]
```diff
@@ -48,51 +48,82 @@ const strapiAuthHelper = AuthHelper(API_URL + "/api");
 
 export const authProvider: AuthProvider = {
```
Comment (BatuhanW) [inline documentation/docs/advanced-tutorials/multi-tenancy/strapi.md:49]: Let's check this line as well.

File: documentation/docs/api-reference/antd/components/auth-page.md [hunk 8]
```diff
@@ -195,9 +219,13 @@ const authProvider: AuthProvider = {
         // You can handle the login process according to your needs.
 
         // If the process is successful.
-        return Promise.resolve();
+        return {
+            status: true,
```
Comment (BatuhanW) [inline documentation/docs/api-reference/antd/components/auth-page.md:223]: Shouldn't it be `success`?

**Review by BatuhanW — CHANGES_REQUESTED** [review]

File: documentation/docs/api-reference/antd/components/auth-page.md (hunk 8 above)
Comment (alicanerdurmaz) [reply documentation/docs/api-reference/antd/components/auth-page.md]: yes

File: documentation/docs/api-reference/core/hooks/auth/useForgotPassword.md [hunk 9]
```diff
@@ -77,36 +79,21 @@ const { mutate: forgotPassword } = useForgotPassword();
...
 
 ```tsx
-const authProvider: AuthProvider = {
-    ...
-    forgotPassword: () => {
-        ...
-        return Promise.resolve(false);
-    }
-}
+const authProvider: AuthBindings = {
```
Comment (BatuhanW) [inline documentation/docs/api-reference/core/hooks/auth/useForgotPassword.md:85]: Let's add import statement.

File: documentation/docs/api-reference/core/providers/auth-provider.md [hunk 10]
```diff
@@ -46,133 +46,151 @@ const App = () => {
...
 ## Methods
 
-Auth provider's methods are expected to return a Promise. So, you can use these async methods to [create auth provider][create-auth-provider-tutorial].
+Auth provider's methods are expected to return a resolved Promise. So, you can use these async methods to [create auth provider][create-auth-provider-tutorial].
 
 An `authProvider` includes the following methods:
 
 ```tsx
-import { AuthProvider } from "@pankod/refine-core";
+import { AuthBindings } from "@pankod/refine-core";
```
Comment (aliemir) [inline documentation/docs/api-reference/core/providers/auth-provider.md:54]:
```suggestion
import type { AuthBindings } from "@pankod/refine-core";
```

File: documentation/docs/api-reference/core/providers/auth-provider.md [hunk 11]
```diff
@@ -196,17 +241,17 @@ These hooks can be used with the `authProvider` authentication and authorization
...
-| checkAuth <div className=" required">Required</div>  | Checks credentials on each route changes  | Authentication still persist          |
-| checkError <div className=" required">Required</div> | Checks if a dataProvider returns an error | Data provider doesn't return an error |
-| getPermissions                                       | Can be use to get user credentials        | Authorization roles accepted          |
-| getUserIdentity                                      | Can be use to get user identity           | User identity available to return     |
-| register                                             | Register user                             | Auth confirms register                |
-| forgotPassword                                       | Can be use to get password reset          | Auth confirms forgot password         |
-| updatePassword                                       | Can be use to get update password         | Auth confirms update password         |
+| Property                                          | Description                               | Resolve condition                     |
```
Comment (aliemir) [inline documentation/docs/api-reference/core/providers/auth-provider.md:244]:
```suggestion
| Property                                          | Description                               | Success condition                     |
```

or smt similar but `resolve condition` is not working for the current status of auth provider

File: documentation/docs/api-reference/core/providers/auth-provider.md [hunk 12]
```diff
@@ -216,12 +261,16 @@ These hooks can be used with the `authProvider` authentication and authorization
...
 [use-permissions]: /docs/api-reference/core/hooks/auth/usePermissions/
 [authenticated]: /docs/api-reference/core/components/auth/authenticated/
 [use-register]: /docs/api-reference/core/hooks/auth/useRegister/
 [use-forgot-password]: /docs/api-reference/core/hooks/auth/useForgotPassword/
 [use-update-password]: /docs/api-reference/core/hooks/auth/useUpdatePassword/
 [create-auth-provider-tutorial]: /docs/tutorial/understanding-authprovider/create-authprovider
+
+```
+
+```
```
Comment (aliemir) [inline documentation/docs/api-reference/core/providers/auth-provider.md:276]:
```suggestion
```

File: documentation/docs/api-reference/core/components/auth/authenticated.md [hunk 13]
```diff
@@ -3,15 +3,15 @@ id: authenticated
...
-It internally uses [`useAuthenticated`][useauthenticated]'s return values (`isSuccess`, `isLoading`, and, `isError`) to provide its functionality.
+It internally uses [`useIsAuthenticated`][use-is-authenticated]'s return values (`data.authenticated`, `data.error`, and, `isLoading`) to provide its functionality.
 
 When:
 
--   `isSuccess` is `true`, it renders to children.
+-   `data.authenticated` is `true`, it renders to children.
 -   `isLoading` is `true`, it renders [`loading`](#loading) prop.
--   `isError` is `true`, it renders [`fallback`](#fallback) prop if provided. Otherwise, it redirects to `/login` page.
+-   `data.error` is `true`, it renders [`fallback`](#fallback) prop if provided. Otherwise, it redirects to `data.redirectTo` page.
```
Comment (aliemir) [inline documentation/docs/api-reference/core/components/auth/authenticated.md:14]: Not true, the code is changed in the source code. `data.error` is a complementary property, has no effect on `Authenticated` directly. 

`fallback` is rendered when `data.authentication` is set to `false`.

File: documentation/docs/api-reference/core/components/auth/authenticated.md [hunk 14]
```diff
@@ -60,6 +60,6 @@ Component to render while checking whether the user is logged in.
 | fallback | Component to render if the user is not logged in. If `undefined`, the page will be redirected to `/login` | `ReactNode` |         |
 | loading  | Component to render while checking whether user is logged in                                              | `ReactNode` |         |
```
Comment (aliemir) [inline documentation/docs/api-reference/core/components/auth/authenticated.md:61]: PropsTable is repeated, one auto and one hand written tables are visible.

File: documentation/docs/tutorial/5-understanding-authprovider/1-create-authprovider.md [hunk 15]
```diff
@@ -13,30 +13,30 @@ This section will show you how to create an auth provider from scratch. We'll us
 1. Create a new file named `authProvider.ts` in `src` folder and add the following code:
 
     ```tsx title="src/authProvider.ts"
-    import { AuthProvider } from "@pankod/refine-core";
+    import { AuthBindings } from "@pankod/refine-core";
```
Comment (aliemir) [inline documentation/docs/tutorial/5-understanding-authprovider/1-create-authprovider.md:16]:
```suggestion
    import type { AuthBindings } from "@pankod/refine-core";
```

File: documentation/docs/tutorial/5-understanding-authprovider/1-create-authprovider.md [hunk 16]
```diff
@@ -143,96 +169,123 @@ mutate({ redirectPath: "/custom-page" });
...
+};
 ```
 
 </details>
 
 <details>
   <summary><strong>How can I customize the error message?</strong></summary>
 
-**refine** automatically displays an error notification when the `login` method rejects the Promise. If you want to customize the error message, you can reject the Promise with an object that has `name` and `message` properties.
+**refine** automatically displays an error notification when the `login` method resolves the Promise with `success: false`. If you want to customize the error message, you can resolve the Promise with an `error` object that has `name` and `message` properties.
```
Comment (aliemir) [inline documentation/docs/tutorial/5-understanding-authprovider/1-create-authprovider.md:204]: ❤️

File: documentation/docs/tutorial/5-understanding-authprovider/0-intro.md [hunk 17]
```diff
@@ -11,49 +11,52 @@ import AuthProviderExamplesLinks from "@site/src/partials/auth-provider-examples
...
-const authProvider: AuthProvider = {
-    login: (params) => Promise,
-    logout: (params) => Promise,
-    checkAuth: (params?) => Promise,
-    checkError: (error) => Promise,
-    ...
+import { AuthBindings } from "@pankod/refine-core";
+
+const authProvider: AuthBindings = {
+    login: (params: any) => Promise.resolve(),
```
Comment (BatuhanW) [inline documentation/docs/tutorial/5-understanding-authprovider/0-intro.md:26]:
```suggestion
    async login: (params: any) => {},
```

Comment (BatuhanW) [reply documentation/docs/tutorial/5-understanding-authprovider/0-intro.md:26]: Let's update these methods, convert them to async and return nothing for mocks across documentation.

File: documentation/docs/api-reference/mui/components/auth-page.md [hunk 18]
```diff
@@ -1002,5 +1057,5 @@ interface FormPropsType extends UseFormProps {
 [forgot-password]: /docs/api-reference/core/providers/auth-provider/#forgotpassword
 [update-password]: /docs/api-reference/core/providers/auth-provider/#updatepassword
 [get-permissions]: /docs/api-reference/core/providers/auth-provider/#getpermissions-
-[check-auth]: /docs/api-reference/core/providers/auth-provider/#checkauth-
+[check-auth]: /docs/api-reference/core/providers/auth-provider/#check-
```
Comment (BatuhanW) [inline documentation/docs/api-reference/mui/components/auth-page.md:1060]: Is this correct?

**Review by aliemir — APPROVED** [review]

File: documentation/docs/api-reference/mantine/components/auth-page.md [hunk 19]
```diff
@@ -24,21 +24,42 @@ const { Button } = RefineMantine;
...
+            redirectTo: "/",
+        };
     },
-    checkAuth: () =>
-        window.__refineAuthStatus ? Promise.resolve() : Promise.reject(),
-    checkError: () => Promise.resolve(),
-    getPermissions: () => Promise.resolve(),
-    getUserIdentity: () => Promise.resolve(),
+    check: async () => ({
+        authenticated: window.__refineAuthStatus ? true : false,
```
Comment (BatuhanW) [inline documentation/docs/api-reference/mantine/components/auth-page.md:58]: Should we add optional redirects?

`redirectTo: window.__refineAuthStatus ? undefined : "/login",`

File: documentation/docs/api-reference/core/providers/auth-provider.md [hunk 20]
```diff
@@ -46,133 +46,151 @@ const App = () => {
...
+`logout` method is used to log out users.
 
-## Optional Methods
+You can use [`useLogout`][use-logout] hook to call `logout` method.
 
-### getPermissions
+### onError <PropTag required />
 
-`getPermissions` method is used to get the user's permissions. It expects to return a Promise.
+`onError` method is called when you get an error response from the API. You can create your own business logic to handle the error such as refreshing the token, logging out the user, etc.
```
Comment (BatuhanW) [inline documentation/docs/api-reference/core/providers/auth-provider.md:152]: Maybe we can also mention sending some notification to the user. Something similar was asked on Discord recently.

File: documentation/docs/api-reference/mui/components/auth-page.md (hunk 18 above)
Comment (aliemir) [reply documentation/docs/api-reference/mui/components/auth-page.md:1060]: When there's an element next to the title (e.g. PropTag for required or deprecated statuses) Docusaurus fails to trim the slug and generates the id with `-` in the end.

File: documentation/docs/api-reference/mantine/components/auth-page.md (hunk 19 above)
Comment (alicanerdurmaz) [reply documentation/docs/api-reference/mantine/components/auth-page.md:58]: yes

**Review by aliemir — APPROVED** [review]

**Review by BatuhanW — APPROVED** [review]

**Review by BatuhanW — APPROVED** [review]

**Review by aliemir — APPROVED** [review]

**Review by aliemir — APPROVED** [review]

**Pull request merged**